src/components/Notification/Notification.js

```diff
@@ -135,7 +135,6 @@ export class NotificationButton extends Component {
           <IconTag
             aria-label={iconDescription}
             className={iconClasses}
-            icon={!icon && !name ? iconClose : icon}
```

Comment:
Found during the test, `icon` prop is for legacy `<Icon>` component and the code here is for new feature in `v10` supporting custom icon via render prop.